kmath-core/src/commonMain/kotlin/scientifik/kmath/operations/KBigInteger.kt

```diff
@@ -0,0 +1,454 @@
+package scientifik.kmath.operations
+
+import kotlin.math.log2
+import kotlin.math.max
+import kotlin.math.min
+import kotlin.math.sign
+
+/*
+ * Kotlin Multiplatform implementation of Big Integer numbers (KBigInteger).
+ * Initial version from https://github.com/robdrynkin/kotlin-big-integer
+ */
+
+@kotlin.ExperimentalUnsignedTypes
+typealias Magnitude = UIntArray
+
+@kotlin.ExperimentalUnsignedTypes
+typealias TBase = ULong
+
+object KBigIntegerRing: Ring<KBigInteger> {
+    override val zero: KBigInteger = KBigInteger.ZERO
+    override val one: KBigInteger = KBigInteger.ONE
+
+    override fun add(a: KBigInteger, b: KBigInteger): KBigInteger = a.plus(b)
+
+    override fun multiply(a: KBigInteger, k: Number): KBigInteger = a.times(k.toLong())
+
+    override fun multiply(a: KBigInteger, b: KBigInteger): KBigInteger = a.times(b)
+
+    operator fun String.unaryPlus(): KBigInteger = KBigInteger(this)!!
+
+    operator fun String.unaryMinus(): KBigInteger = -KBigInteger(this)!!
+}
+
+
+@kotlin.ExperimentalUnsignedTypes
+class KBigInteger(sign: Int = 0, magnitude: Magnitude = Magnitude(0)):
+        RingElement<KBigInteger, KBigInteger, KBigIntegerRing>, Comparable<KBigInteger> {
```

Comment:
It does not make sense to make it RingElement since all operations are already class members.